### ISSUE TITLE
Error on triplet loss layer

### ISSUE THREAD

**Issue by ghost** [chainer/chainer#2473]
It seems there is an error with [the triplet loss layer](https://github.com/pfnet/chainer/blob/master/chainer/functions/loss/triplet.py)  when using build_computational_graph.

Assume I have Variable **`loss`** that is a result of the triplet loss layer (**`with Batch size 1`**).
When I use **`g = c.build_computational_graph(loss)`**, it returns 

```
chainer.utils.type_check.InvalidType: 
Invalid operation is performed in: GetItem (Forward)
Expect: in_types[0].ndim >= 1
Actual: 0 < 1
```

I guess this error is due to **`0 ndim`** of the Variable loss.
So, I modified [this line of the triplet loss layer](https://github.com/pfnet/chainer/blob/master/chainer/functions/loss/triplet.py#L41 )
from
**`return xp.array(loss, dtype=numpy.float32)`**,
to
**`return xp.array([loss], dtype=numpy.float32)`**,

which returns **`1 ndim`** for the Variable loss. (instead of **`0 ndim`**)

I can draw a computational graph with the above approach, but I wonder whether the approach is not harmful to the entire network in terms of backpropagation.
(I use Chainer in Windows 10 with Anaconda 3 and Python 3.5.2)

Thanks.

**Comment by soramichi** [comment]
I think this issue is related to #2077.

`loss` is calculated in the previous line as `loss = xp.sum(self.dist_hinge) / N`, and the thing is that ndim of the return value of `xp.sum` differs depending on if `xp` is `numpy` or `cupy`.

```
>>> cupy.sum(cupy.array([1,2,3]))
array(6)
>>> numpy.sum(numpy.array([1,2,3]))
6
>>>
```
